src/mlpack/core/optimizers/parallel_sgd/sparse_test_function.hpp

```diff
@@ -27,21 +27,59 @@ class SparseTestFunction
 {
  public:
   //! Set members in the default constructor.
-  SparseTestFunction();
+  SparseTestFunction()
+  {
+    intercepts = arma::vec("20 12 15 100");
+    bi = arma::vec("-4 -2 -3 -8");
+  }
 
   //! Return 4 (the number of functions).
   size_t NumFunctions() const { return 4; }
 
+  //! Return 4 (the number of features).
+  size_t NumFeatures() const { return 4; }
+
   //! Get the starting point.
   arma::mat GetInitialPoint() const { return arma::mat("0; 0; 0; 0;"); }
 
+  //! Evaluate all the functions.
+  double Evaluate(const arma::mat& coordinates, const size_t i) const
+  {
+    return coordinates[i] * coordinates[i] + bi[i] * coordinates[i] +
+      intercepts[i];
+  }
+
   //! Evaluate a function.
```

Comment:
Minor - I think this comment and the one on the overload above should be swapped?